content/tutorial-x-ray-image-processing.md

```diff
@@ -0,0 +1,571 @@
+
+axes[0].set_title('Original')
+axes[0].imshow(xray_image, cmap='gray')
+axes[1].set_title('Laplacian-Gaussian (edges)')
+axes[1].imshow(xray_image_laplace_gaussian, cmap='gray')
+for i in axes:
+    i.axis('off')
+plt.show()
+```
+
+### The Gaussian gradient magnitude method
+
+Another method for edge detection that can be useful is the
+[Gaussian](https://en.wikipedia.org/wiki/Normal_distribution) (gradient) filter.
+It computes the multidimensional gradient magnitude with Gaussian derivatives
+and helps by remove
+[high-frequency](https://www.cs.cornell.edu/courses/cs6670/2011sp/lectures/lec02_filter.pdf)
+image components.
+
++++
+
+**1.** Call [`scipy.ndimage.gaussian_gradient_magnitude()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.gaussian_gradient_magnitude.html)
+with a sigma (scalar) parameter (for standard deviations; you'll use `2` in the
+example below):
+
+```{code-cell} ipython3
+x_ray_image_gaussian_gradient = ndimage.gaussian_gradient_magnitude(xray_image, sigma=2)
+```
+
+**2.** Display the original X-ray and the one with the Gaussian gradient filter:
+
+```{code-cell} ipython3
+fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(10, 10))
+
+axes[0].set_title('Original')
+axes[0].imshow(xray_image, cmap='gray')
+axes[1].set_title('Gaussian gradient (edges)')
+axes[1].imshow(x_ray_image_gaussian_gradient, cmap='gray')
+for i in axes:
+    i.axis('off')
+plt.show()
+```
+
+### The Sobel-Feldman operator (the Sobel filter)
+
+To find regions of high spatial frequency (the edges or the edge maps) along the
+horizontal and vertical axes of a 2D X-ray image, you can use the
+[Sobel-Feldman operator (Sobel filter)](https://en.wikipedia.org/wiki/Sobel_operator)
+technique. The Sobel filter applies two 3x3 kernel matrices — one for each axis
+— onto the X-ray through a [convolution](https://en.wikipedia.org/wiki/Kernel_(image_processing)#Convolution).
+Then, these two points (gradients) are combined using the
+[Pythagorean theorem](https://en.wikipedia.org/wiki/Pythagorean_theorem) to
+produce a gradient magnitude.
+
++++
+
+**1.** Use the Sobel filters — ([`scipy.ndimage.sobel()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.sobel.html))
+— on x- and y-axes of the X-ray. Then, calculate the distance between `x` and
+`y` (with the Sobel filters applied to them) using the
+[Pythagorean theorem](https://en.wikipedia.org/wiki/Pythagorean_theorem) and
+NumPy's [`np.hypot()`](https://numpy.org/doc/stable/reference/generated/numpy.hypot.html)
+to obtain the magnitude. Finally, normalize the rescaled image for the pixel
+values to be between 0 and 255. 
+
+    [Image normalization](https://en.wikipedia.org/wiki/Normalization_%28image_processing%29)
+    follows the `output_channel = 255.0 * (input_channel - min_value) / (max_value - min_value)`
+    [formula](http://dev.ipol.im/~nmonzon/Normalization.pdf). Because you're
+    using a grayscale image, you need to normalize just one channel.
```

Comment:
```suggestion
[Image normalization](https://en.wikipedia.org/wiki/Normalization_%28image_processing%29)
follows the `output_channel = 255.0 * (input_channel - min_value) / (max_value - min_value)`
[formula](http://dev.ipol.im/~nmonzon/Normalization.pdf). Because you're
using a grayscale image, you need to normalize just one channel.
```
The indentation here causes this to be interpreted [as a literal block](https://59-248354526-gh.circle-artifacts.com/0/site/_build/html/content/tutorial-x-ray-image-processing.html#the-sobel-feldman-operator-the-sobel-filter).

Note that MyST supports admonitions just like rST:
````suggestion
```{note}
[Image normalization](https://en.wikipedia.org/wiki/Normalization_%28image_processing%29)
follows the `output_channel = 255.0 * (input_channel - min_value) / (max_value - min_value)`
[formula](http://dev.ipol.im/~nmonzon/Normalization.pdf). Because you're
using a grayscale image, you need to normalize just one channel.
```
````
unfortunately since this is a MyST-specific feature, they don't render properly in `.ipynb` files, only the html.

Reply:
Good point